Artifacts/windows-clone-git-repo/StartChocolatey.ps1

```diff
@@ -16,61 +16,201 @@ Param(
+{
+    [CmdletBinding()]
+    param(
+        [int] $length = 43
+    )
+
+    $sourceData = $null
+    33..126 | % { $sourceData +=,[char][byte]$_ }
+
+    1..$length | % { $tempPassword += ($sourceData | Get-Random) }
 
     return $tempPassword
 }
 
-# Ensure Powershell 3 or more is installed.
-if ($PSVersionTable.PSVersion.Major -lt 3)
+function Add-LocalAdminUser
 {
-    Write-Error "Prior to running this artifact, ensure you have Powershell 3 or higher installed."
-    [System.Environment]::Exit(1)
+    [CmdletBinding()]
+    param(
+        [string] $UserName,
+        [string] $Password,
+        [string] $Description = 'DevTestLab artifact installer',
+        [switch] $Overwrite = $true
+    )
+    
+    if ($Overwrite)
+    {
+        Remove-LocalAdminUser -UserName $UserName
+    }
+
+    $computer = [ADSI]"WinNT://$env:ComputerName"
+    $user = $computer.Create("User", $UserName)
+    $user.SetPassword($Password)
+    $user.Put("Description", $Description)
+    $user.SetInfo()
+
+    $group = [ADSI]"WinNT://$env:ComputerName/Administrators,group"
+    $group.add("WinNT://$env:ComputerName/$UserName")
+    
+    return $user
 }
 
-$ascii=$NULL;For ($a=33;$a -le 126;$a++) {$ascii+=,[char][byte]$a }
+function Remove-LocalAdminUser
+{
+    [CmdletBinding()]
+    param(
+        [string] $UserName
+    )
 
-$userName = "artifactInstaller"
-$password = Get-TempPassword -length 43 -sourcedata $ascii
+    if ([ADSI]::Exists('WinNT://./' + $UserName))
+    {
+        $computer = [ADSI]"WinNT://$env:ComputerName"
+        $computer.Delete('User', $UserName)
+        try
+        {
+            gwmi win32_userprofile | ? { $_.LocalPath -like "*$UserName*" -and -not $_.Loaded } | % { $_.Delete() | Out-Null }
+        }
+        catch
+        {
+            # Ignore any errors, specially with locked folders/files. It will get cleaned up at a later time, when another artifact is installed.
+        }
+    }
+}
 
-$cn = [ADSI]"WinNT://$env:ComputerName"
+function Set-LocalAccountTokenFilterPolicy
+{
+    [CmdletBinding()]
+    param(
+        [int] $Value = 1
+    )
+
+    $oldValue = 0
+
+    $regPath ='HKLM:\Software\Microsoft\Windows\CurrentVersion\Policies\System'
+    $policy = Get-ItemProperty -Path $regPath -Name LocalAccountTokenFilterPolicy -ErrorAction SilentlyContinue
+
+    if ($policy)
+    {
+        $oldValue = $policy.LocalAccountTokenFilterPolicy
+    }
+
+    if ($oldValue -ne $Value)
+    {
+        Set-ItemProperty -Path $regPath -Name LocalAccountTokenFilterPolicy -Value $Value
+    }
+
+    return $oldValue
+}
+
+function Invoke-GitEnlister
+{
+    [CmdletBinding()]
+    param(
+        [string] $UserName,
+        [string] $Password,
+        [array] $ArgumentList
+    )
 
-# Create user
-$user = $cn.Create("User", $userName)
-$user.SetPassword($password)
-$user.SetInfo()
-$user.description = "DevTestLab artifact installer"
-$user.SetInfo()
+    $secPassword = ConvertTo-SecureString -String $Password -AsPlainText -Force

```

Comment:
I will make this change when I consolidate the many copies of startChocolatey.ps1.